test/integration/buildruns_to_taskruns_test.go

```diff
@@ -393,4 +393,62 @@ var _ = Describe("Integration tests BuildRuns and TaskRuns", func() {
 			Expect(err).To(BeNil(), fmt.Sprintf("failed to get desired reason; expected %s, got %s", expectedReason, actualReason))
 		})
 	})
+
+	Context("when a buildrun is created and the taskrun deleted before completion", func() {
+
+		BeforeEach(func() {
+			buildSample = []byte(test.BuildCBSMinimal)
+			buildRunSample = []byte(test.MinimalBuildRun)
+		})
+
+		It("should reflect a Failed reason", func() {
+
+			Expect(tb.CreateBuild(buildObject)).To(BeNil())
+
+			buildObject, err = tb.GetBuildTillValidation(buildObject.Name)
+			Expect(err).To(BeNil())
+
+			Expect(tb.CreateBR(buildRunObject)).To(BeNil())
+
+			_, err = tb.GetBRTillStartTime(buildRunObject.Name)
+			Expect(err).To(BeNil())
+
+			tr, err := tb.GetTaskRunFromBuildRun(buildRunObject.Name)
+			Expect(err).To(BeNil())
+
+			tb.DeleteTR(tr.Name)
+
+			expectedReason := fmt.Sprintf("taskRun %s doesn't exist", tr.Name)
+			actualReason, err := tb.GetBRTillDesiredReason(buildRunObject.Name, expectedReason)
+			Expect(err).To(BeNil(), fmt.Sprintf("failed to get desired reason; expected %s, got %s", expectedReason, actualReason))
+		})
+	})
+
+	Context("when a buildrun is created and the taskrun deleted after successful completion", func() {
+
+		It("should reflect a Success reason", func() {
+			WithCustomClusterBuildStrategy([]byte(test.ClusterBuildStrategyNoOp), func() {
+				_, _, buildRunObject := setupBuildAndBuildRun([]byte(test.BuildCBSMinimal), []byte(test.MinimalBuildRun), STRATEGY+tb.Namespace+"custom")
+
+				_, err = tb.GetBRTillCompletion(buildRunObject.Name)
+				Expect(err).To(BeNil())
+
+				reason, err := tb.GetBRReason(buildRunObject.Name)
+				Expect(err).To(BeNil())
+				Expect(reason).To(Equal("Succeeded"))
+
+				tr, err := tb.GetTaskRunFromBuildRun(buildRunObject.Name)
+				Expect(err).To(BeNil())
+
+				tb.DeleteTR(tr.Name)
+
+				// in a test case, it is hard to verify that something (marking the BuildRun failed is not happening)
+				time.Sleep(5 * time.Second)
```

Comment:
this one doesnt looks good. I think this test case is missing the following:
- assert a TaskRun was completed before deleting it.

Instead of the sleep, can you try to retrieve the TaskRun one more time and ensure you get a `NotFound`, then assert that a BuildRun continue to have a `Succeeded` reason`?

Reply:
Yeah, also possible. There is no correct way to verify that something does not happen because it eventually can still happen. Changed it to what you suggest.